pkg/server/events.go

```diff
@@ -79,11 +92,33 @@ func (em *eventMonitor) start() (<-chan struct{}, error) {
 			select {
 			case e := <-em.ch:
 				logrus.Debugf("Received containerd event timestamp - %v, namespace - %q, topic - %q", e.Timestamp, e.Namespace, e.Topic)
-				em.handleEvent(e)
+				any, err := typeurl.UnmarshalAny(e.Event)
+				if err != nil {
+					logrus.WithError(err).Errorf("Failed to convert event envelope %+v", e)
+					break
+				}
+				if cID, backOffIng := em.isBackOffIng(any); backOffIng {
+					em.enBackOff(cID, any)
+					break
+				}
+				if cID, err := em.handleEvent(any); err != nil && cID != "" {
+					em.enBackOff(cID, any)
+				}
 			case err := <-em.errCh:
 				logrus.WithError(err).Error("Failed to handle event stream")
 				close(closeCh)
 				return
+			case cID := <-em.backOffExpire:
+				for {
+					any := em.deBackOff(cID)
+					if any == nil {
+						break
+					}
+					if _, err := em.handleEvent(any); err != nil && cID != "" {
```

Comment:
`cID != ""`? You don't even set it based on the return value.

Reply:
will do